main.js

```diff
@@ -30,3 +31,149 @@ const menuLinks = mobile_menu.querySelectorAll("a[href^='#']");
+    popupTitle: 'Keeping track of hundreds of components',
+  },
+];
+
+const projectsWrapper = document.querySelector('.projects__wrapper');
+const popupWindow = document.getElementById('popup__window');
+const popupOverlay = document.getElementById('popup__overlay');
+const popupImage = document.getElementById('popup__image');
+const popupTitle = document.getElementById('popup__title');
+const popupTags = document.getElementById('popup__card__tags');
+const popupDescription = document.getElementById('popup__description');
+const liveLinkButton = document.getElementById('popup__live__link__button');
+const sourceLinkButton = document.getElementById('popup__source__link__button');
+const closeButton = document.getElementById('popup__close__button');
+
+function createProjectCard(project) {
+  const card = document.createElement('article');
+  card.className = 'project__card';
+  card.innerHTML = `
+          <div class="project__card__description">
+            <h3 class="project__card__title">${project.name}</h3>
+            <ul class="project__card__tags">
+              ${project.techTags.map((tech) => `<li>${tech}</li>`).join('')}
+            </ul>
+            <button class="project__card__button" type="button">See project</button>
+          </div>
+          <img src="${project.imgMobile}" alt="${project.name}" class="project__card__image">
+        `;
+
+  card.querySelector('.project__card__button').addEventListener('click', () => {
+    // eslint-disable-next-line no-use-before-define
+    showPopup(project);
+  });
+
+  return card;
+}
+
+function showPopup(project) {
+  const screenWidth = window.innerWidth;
+  if (screenWidth > 768) {
+    popupImage.src = project.imgDesktop;
+  } else {
+    popupImage.src = project.imgMobile;
+  }
+  popupTitle.textContent = project.popupTitle || project.name;
+  popupTags.classList.add('popup__tags');
+  let tagsHTML = '';
+
+  if (screenWidth < 768) {
+    tagsHTML = `
+      <li>Ruby on Rails</li>
+      <li>CSS</li>
+      <li>JavaScript</li>
+    `;
+  } else {
+    tagsHTML = `
+      <li>Codekit</li>
+      <li>GitHub</li>
+      <li>JavaScript</li>
+      <li>Bootstrap</li>
+      <li>Terminal</li>
+      <li>Codepen</li>
+    `;
+  }
+
+  popupTags.innerHTML = tagsHTML;
+
+  popupDescription.textContent = project.description;
+  liveLinkButton.href = project.liveLink;
+  sourceLinkButton.href = project.sourceLink;
+  popupWindow.style.display = 'flex';
+  popupOverlay.style.display = 'block';
+  document.body.classList.add('popup-open');
+}
+
+function hidePopup() {
+  popupWindow.style.display = 'none';
+  popupOverlay.style.display = 'none';
+  document.body.classList.remove('popup-open');
+  popupTags.classList.remove('popup__tags');
+}
+
+projects.forEach((project) => {
+  const card = createProjectCard(project);
+  projectsWrapper.appendChild(card);
+});
+
+closeButton.addEventListener('click', hidePopup);
```

Comment:
- [ ] I can see that you attached an event listener to a button to chose the popup, but I cant see the button in your popup UI. Kindly check it out 👍🏽 

your design 👇🏽 
![image](https://github.com/rudicarrilloypr/portfolio/assets/63473584/76256ae1-8b05-491c-a228-84c5dc65e025)

Figma design 👇🏽 
![image](https://github.com/rudicarrilloypr/portfolio/assets/63473584/b9bcfa51-ba49-4c28-ad6b-c9d8052e510c)

Reply:
@vickymarz thank you very much, to be honest, I'm not sure how to fix it because it is declared to function, can you give me any advice to fix this problem?

Reply:
@rudicarrilloypr get the close button to display first on the UI, before ching the functionality